react-paw-mailmerge/src/components/FileUpload.tsx

```diff
@@ -11,10 +11,33 @@ const FileUpload: React.FC = () => {
     setSelectedFile(file || null);
   };
 
+  const validateFile = (file: File): string[] => {
+    const errors: string[] = [];
+    const type = file.type;
+    const size_limit = 10 * 1024 * 1024;  // file.size is in bytes. 10 mb = 10 * 1024 * 1024 bytes
+
+    if (type !== 'text/csv') {
+      errors.push(`Expected file with MIME type text/csv, got ${type}.`);
```

Comment:
Good descriptive error message. Should we include the file name as well?

Reply:
That's a good idea. I will update this.